Sources/TuistKit/Generator/TuistGeneratorModelLoader.swift

```diff
@@ -0,0 +1,215 @@
+        let name: String = try json.get("name")
+        let tool: String? = try? json.get("tool")
+        let order = TuistKit.TargetAction.Order(rawValue: try json.get("order"))!
+        let pathString: String? = try? json.get("path")
+        let path = pathString.map { AbsolutePath($0, relativeTo: path) }
+        let arguments: [String] = try json.get("arguments")
+        return TargetAction(name: name, order: order, tool: tool, path: path, arguments: arguments)
+    }
+}
+
+extension TuistKit.CoreDataModel {
+    static func from(json: JSON, path: AbsolutePath, fileHandler: FileHandling) throws -> TuistKit.CoreDataModel {
+        let pathString: String = try json.get("path")
+        let modelPath = path.appending(RelativePath(pathString))
+        if !fileHandler.exists(modelPath) {
+            throw GraphLoadingError.missingFile(modelPath)
+        }
+        let versions: [AbsolutePath] = path.glob("*.xcdatamodel")
+        let currentVersion: String = try json.get("current_version")
+        return CoreDataModel(path: modelPath, versions: versions, currentVersion: currentVersion)
+    }
+}
+
+extension TuistKit.Headers {
+    static func from(json: JSON, path: AbsolutePath, fileHandler: FileHandling) throws -> TuistKit.Headers {
+        let publicString: String? = try? json.get("public")
+        let `public` = publicString.map { path.glob($0) } ?? []
+        let privateString: String? = try? json.get("private")
+        let `private` = privateString.map { path.glob($0) } ?? []
+        let projectString: String? = try? json.get("project")
+        let project = projectString.map { path.glob($0) } ?? []
+        return Headers(public: `public`, private: `private`, project: project)
+    }
+}
+
+extension TuistKit.Dependency {
+    static func from(json: JSON, path: AbsolutePath, fileHandler: FileHandling) throws -> TuistKit.Dependency {
+        let type: String = try json.get("type")
+        switch type {
+        case "target":
+            return .target(name: try json.get("name"))
+        case "project":
+            let target: String = try json.get("target")
+            let path: String = try json.get("path")
+            return .project(target: target, path: RelativePath(path))
+        case "framework":
+            let path: String = try json.get("path")
+            return .framework(path: RelativePath(path))
+        case "library":
+            let path: String = try json.get("path")
+            let publicHeaders: String = try json.get("public_headers")
+            let swiftModuleMap: RelativePath? = json.get("swift_module_map").map { RelativePath($0) }
+            return .library(path: RelativePath(path),
+                            publicHeaders: RelativePath(publicHeaders),
+                            swiftModuleMap: swiftModuleMap)
+        default:
+            throw TuistGeneratorModelLoaderError.malformedManifest
+        }
+    }
+}
```

Comment:
I'm in 2 minds about this - 

One one hand it's nice having all the conversions in one place to clearly indicate this is where the transformation of the manifests > models is taking place.

On the other hand it might be too much in one place.

The motivation is to decouple the JSON format knowledge from the model objects to allow seamlessly moving them across to `TuistGenerator`

Reply:
I'd prefer this in one place. It will make it easier when we want to swap to use `Decodable`.